airbyte-integrations/connectors/source-hubspot/source_hubspot/components.py

```diff
@@ -137,3 +139,62 @@ def extract_records(self, response: requests.Response) -> Iterable[Mapping[str,
                             version["property"] = property_name
                             version[self.entity_primary_key] = primary_key
                             yield version | additional_keys
+
+
+class EngagementsHttpRequester(HttpRequester):
+    recent_api_total_records_limit = 10000
+    recent_api_last_days_limit = 29
+
+    recent_api_path = "/engagements/v1/engagements/recent/modified"
+    all_api_path = "/engagements/v1/engagements/paged"
+
+    _use_recent_api = None
+
+    def should_use_recent_api(self, stream_slice: StreamSlice) -> bool:
+        if self._use_recent_api is not None:
+            return self._use_recent_api
+
+        # Recent engagements API returns records updated in the last 30 days only. If start time is older All engagements API should be used
+        if int(stream_slice["start_time"]) >= int((pendulum.now() - timedelta(days=self.recent_api_last_days_limit)).timestamp() * 1000):
```

Comment:
We should avoid using `pendulum` in new places and instead use `ab_datetime_now()` (or something equivalent). Because we plan to make this manifest-only which will not support the `pendulum` library, avoiding new usages makes our future migration easier.

Reply:
added `ab_datetime_now` with `DateTimeParser`, removed `pendulum`